File: sdk/src/models/event.ts
import { DateTime } from 'luxon';

export class PhEvent {
    id: number | null;
    name: string;
    author: string | null;
    date: DateTime;
    location: string | null;
    nexusId: string | null;
    registrationUrl: string | null;
    displayText: string | null = null;
    displayTextAppliance: boolean = false;

    // This fucking frontend sucks so much
    // that it isn't parsed the same way in mercure & api
    // so fuck that sshit i dont care i need it working for tomorrow
    // this will stay like this until the rewrite
    display_text: string | null = null;
    display_text_appliance: boolean = false;
    registration_url: string | null = null;

    amtImagesHandtaken: number = 0;
    amtImagesUnattended: number = 0;

    public constructor(data: Record<string, any>) {
        this.id = data.id;
        this.name = data.name;
        this.author = data.author;
        this.date = DateTime.fromISO(data.date);
        this.location = data.location;
        this.nexusId = data.nexus_id;
        this.registrationUrl = data.registration_url;
        this.displayText = data.display_text;
        this.displayTextAppliance = data.display_text_appliance;
        this.amtImagesHandtaken = data.amt_images_handtaken;
        this.amtImagesUnattended = data.amt_images_unattended;
    }

    static fromJson(json: Record<string, any> | null): PhEvent | null {
        if (!json) {
            return null;
        }

        return new PhEvent(json);
    }

    asJson(): Record<string, any> {
        return {
            id: this.id,
            name: this.name,
            author: this.author,
            date: this.date.toISO(),
            location: this.location,
            nexus_id: this.nexusId,
            registration_url: this.registrationUrl,
            display_text: this.displayText,
            display_text_appliance: this.displayTextAppliance,
        };
    }
}

File: backend/models/event.go
package models

import (
	"time"
)

type Event struct {
	Id       int64     `db:"id" json:"id"`
	Name     string    `db:"name" json:"name"`
	Author   string    `db:"author" json:"author"`
	Date     time.Time `db:"date" json:"date"`
	Location string    `db:"location" json:"location"`

	NexusId             JsonnableNullstring `db:"nexus_id" json:"nexus_id"`
	UserRegistrationUrl JsonnableNullstring `db:"registration_url" json:"registration_url"`

	DisplayText          JsonnableNullstring `db:"display_text" json:"display_text"`
	DisplayTextAppliance bool                `db:"display_text_appliance" json:"display_text_appliance"`

	AmtImageHandtaken  int `db:"amt_images_handtaken" json:"amt_images_handtaken"`
	AmtImageUnattended int `db:"amt_images_unattended" json:"amt_images_unattended"`
}

func (e Event) AsJson() map[string]any {
	data := map[string]any{
		"id":                     e.Id,
		"name":                   e.Name,
		"author":                 e.Author,
		"date":                   e.Date.Format(time.RFC3339),
		"location":               e.Location,
		"nexus_id":               e.NexusId,
		"user_registration_url":  e.UserRegistrationUrl,
		"display_text":           e.DisplayText,
		"display_text_appliance": e.DisplayTextAppliance,
		"amt_images_handtaken":   e.AmtImageHandtaken,
		"amt_images_unattended":  e.AmtImageUnattended,
	}

	return data
}

type Picture struct {
	Id         string              `db:"id" json:"id"`
	TakenAt    time.Time           `db:"taken_at" json:"taken_at"`
	Unattended bool                `db:"unattended" json:"unattended"`
	Filename   string              `db:"filename" json:"-"`
	EventId    int64               `db:"event_id" json:"-"`
	NexusId    JsonnableNullstring `db:"nexus_id" json:"nexus_id"`
}

func (p Picture) AsJson() map[string]any {
	return map[string]any{
		"id":       p.Id,
		"filename": p.Filename,
		"taken_at": p.TakenAt.Format(time.RFC3339),
	}
}
